test/session_tests/DocumentStoreTest.ts

```diff
@@ -182,6 +182,16 @@ describe('Document store test', () => {
       expect(product['@metadata']['Raven-Node-Type']).to.not.exist;  
       expect(product['@metadata']['@collection']).to.equals('@empty');      
     });
+
+    it('shouldn\'t set id on plain object without id attribute defined', async () => {
+      let product: Product = <Product>{name: "New Product"};    
+
+      session = store.openSession({requestExecutor});      
+      await session.store<Product>(product);
+      await session.saveChanges();
+
+      expect(product.id).to.be.empty;      
```

Comment:
@ayende C# client also removes identity property from JSON when SaveChanges is being called. What behavior should be expected in NodeJS and any other client in regards to that?